python/whylogs/api/whylabs/session/session_manager.py

```diff
@@ -259,31 +61,83 @@ def is_active() -> bool:
         return SessionManager.get_instance() is not None
 
 
-def init(session_type: Union[SessionType, str] = SessionType.LOCAL, reinit: bool = False) -> None:
+def init(
+    reinit: bool = False,
+    allow_anonymous: bool = True,
+    allow_local: bool = False,
+    whylabs_api_key: Optional[str] = None,
+    default_dataset_id: Optional[str] = None,
+) -> None:
```

Comment:
This is a breaking change in the signature of why.init

Is it difficult to maintain the old session_type parameter so we don't break the previous callers that had this:
`why.init(session_type='whylabs_anonymous')`

A known impact would be in the LangKit example notebooks, e.g. we would need to update the init if we update the whylogs dependency. See: https://github.com/whylabs/langkit/blob/main/langkit/examples/Intro_to_Langkit.ipynb

Reply:
LangKit pins the whylogs dependency, so we can update the init in the notebook examples as we update the dependency on whylogs.

I wanted to call attention to the breaking change so it was understood and intentional and we could reason about the risks and mitigations.